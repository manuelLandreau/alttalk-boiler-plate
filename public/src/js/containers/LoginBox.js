import React from 'react'
import {connect} from 'react-redux'
import {login, register, clearErrors} from '../actions/authActions';
import {openModal, closeModal, switchToLogin, switchToRegister} from '../actions/uiActions';
import Modal from '../components/Modal';
import LoginForm from '../components/LoginForm';
import RegisterForm from '../components/RegisterForm';
import UserIcon from '../icons/UserIcon';

@connect((store) => {
    return {
        isOpen: store.ui.isOpen,
        tabSwitch: store.ui.tabSwitch,
        error: store.auth.error.message
    };
})
export default class LoginBox extends React.Component {

    constructor(props) {
        super(props);
    }

    static componentWillUpdate(nextProps, nextState) {
        console.log(nextProps !== nextState);
        return nextProps !== nextState;
    }

    handleLogin = (values) => {
        this.props.dispatch(login(values.username, values.password));
    };

    handleRegister = (values) => {
        this.props.dispatch(register(values.email, values.username, values.password));
    };

    handleOpenModal() {
        this.props.dispatch(openModal());
    }

    handleCloseModal() {
        this.props.dispatch(closeModal());
    }

    switchTabLogin() {
        this.props.dispatch(clearErrors());
        this.props.dispatch(switchToLogin());
    }

    switchTabRegister() {
        this.props.dispatch(clearErrors());
        this.props.dispatch(switchToRegister());
    }

    render() {
        const {isOpen, tabSwitch, error} = this.props;

        return (
            <div>
                <button class="btn" onClick={this.handleOpenModal.bind(this)}>
                    <UserIcon width="16" height="16" />
                    <span class="tiny-hidden"> Sign in / Sign up</span>
                </button>
                <div>
                    <Modal
                        isModalOpen={isOpen}
                        closeModal={this.handleCloseModal.bind(this)}
                        style={{}}
                    >
                        <div>
                            <button class="fr btn" onClick={this.handleCloseModal.bind(this)}>&times;</button>
                        </div>
                        <div class="pam">
                            <button class="btn" onClick={this.switchTabLogin.bind(this)}>Login</button>
                            <button class="btn" onClick={this.switchTabRegister.bind(this)}>Register</button>
                        </div>
                        <div style={{display: tabSwitch ? 'inherit' : 'none'}}>
                            <LoginForm onSubmit={this.handleLogin.bind(this)} message={error}/>
                        </div>
                        <div style={{display: tabSwitch ? 'none' : 'inherit'}}>
                            <RegisterForm onSubmit={this.handleRegister.bind(this)} message={error}/>
                        </div>
                    </Modal>
                </div>
            </div>
        );
    }
}
