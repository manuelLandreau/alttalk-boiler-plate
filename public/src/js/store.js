import {applyMiddleware, createStore} from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import {composeWithDevTools} from 'redux-devtools-extension';
import createBrowserHistory from 'history/createBrowserHistory';
import {routerMiddleware} from 'react-router-redux'
import reducers from './reducers';

export const history = createBrowserHistory();
const middleware = routerMiddleware(history);

export const store = createStore(
    reducers,
    {},
    composeWithDevTools(applyMiddleware(middleware, promise(), thunk, logger()))
);