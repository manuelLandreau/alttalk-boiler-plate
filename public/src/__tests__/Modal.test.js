import React from 'react';
import Modal from '../js/components/Modal';
import LoginBox from '../js/containers/LoginBox';
import renderer from 'react-test-renderer';
import store from '../js/store';
import {shallow} from 'enzyme';

test('Modal simple Snapshot test', () => {
    const component = renderer.create(
        <Modal isModalOpen={false} closeModal={() => false} style={{}}></Modal>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Modal click test', () => {
    const component = renderer.create(
        <LoginBox store={store}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.handleOpenModal;
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.handleCloseModal;
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Registration test', () => {
    const component = renderer.create(
        <LoginBox store={store}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.handleOpenModal;
    tree.props.handleFormChange;
    tree.props.handleRegister;
});