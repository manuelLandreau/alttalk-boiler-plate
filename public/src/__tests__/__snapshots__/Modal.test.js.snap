// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal click test 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    -&gt;
    <span
      className="hidden--thumb"
    >
      Sign in / Sign up
    </span>
  </button>
  <div
    className="relative"
  >
    <div
      style={
        Object {
          "display": "none",
          "height": "100%",
          "left": 0,
          "overflow": "auto",
          "position": "fixed",
          "top": 0,
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
      <div
        onClick={[Function]}
      />
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 3,
            "boxShadow": "0 20px 30px rgba(0, 0, 0, 0.2)",
            "boxSizing": "border-box",
            "margin": "40px auto",
            "padding": "15px 20px",
            "position": "relative",
            "textAlign": "left",
            "width": 500,
            "zIndex": 2,
          }
        }
      >
        <div
          className="grid--right pb40"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
          >
            Login
          </button>
          <button
            onClick={[Function]}
          >
            Register
          </button>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "inherit",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="login_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="login_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="login_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="login_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="register_email"
            >
              Email
            </label>
            <label
              className="grid__cell"
              htmlFor="register_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="register_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="register_email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal click test 2`] = `
<div>
  <button
    onClick={[Function]}
  >
    -&gt;
    <span
      className="hidden--thumb"
    >
      Sign in / Sign up
    </span>
  </button>
  <div
    className="relative"
  >
    <div
      style={
        Object {
          "display": "none",
          "height": "100%",
          "left": 0,
          "overflow": "auto",
          "position": "fixed",
          "top": 0,
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
      <div
        onClick={[Function]}
      />
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 3,
            "boxShadow": "0 20px 30px rgba(0, 0, 0, 0.2)",
            "boxSizing": "border-box",
            "margin": "40px auto",
            "padding": "15px 20px",
            "position": "relative",
            "textAlign": "left",
            "width": 500,
            "zIndex": 2,
          }
        }
      >
        <div
          className="grid--right pb40"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
          >
            Login
          </button>
          <button
            onClick={[Function]}
          >
            Register
          </button>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "inherit",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="login_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="login_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="login_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="login_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="register_email"
            >
              Email
            </label>
            <label
              className="grid__cell"
              htmlFor="register_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="register_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="register_email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal click test 3`] = `
<div>
  <button
    onClick={[Function]}
  >
    -&gt;
    <span
      className="hidden--thumb"
    >
      Sign in / Sign up
    </span>
  </button>
  <div
    className="relative"
  >
    <div
      style={
        Object {
          "display": "none",
          "height": "100%",
          "left": 0,
          "overflow": "auto",
          "position": "fixed",
          "top": 0,
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
      <div
        onClick={[Function]}
      />
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 3,
            "boxShadow": "0 20px 30px rgba(0, 0, 0, 0.2)",
            "boxSizing": "border-box",
            "margin": "40px auto",
            "padding": "15px 20px",
            "position": "relative",
            "textAlign": "left",
            "width": 500,
            "zIndex": 2,
          }
        }
      >
        <div
          className="grid--right pb40"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
          >
            Login
          </button>
          <button
            onClick={[Function]}
          >
            Register
          </button>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "inherit",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="login_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="login_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="login_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="login_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="register_email"
            >
              Email
            </label>
            <label
              className="grid__cell"
              htmlFor="register_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="register_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="register_email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal simple Snapshot test 1`] = `
<div
  style={
    Object {
      "display": "none",
      "height": "100%",
      "left": 0,
      "overflow": "auto",
      "position": "fixed",
      "top": 0,
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "height": "100%",
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "width": "100%",
      }
    }
  />
  <div
    onClick={[Function]}
  />
  <div
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 3,
        "boxShadow": "0 20px 30px rgba(0, 0, 0, 0.2)",
        "boxSizing": "border-box",
        "margin": "40px auto",
        "padding": "15px 20px",
        "position": "relative",
        "textAlign": "left",
        "width": 500,
        "zIndex": 2,
      }
    }
  />
</div>
`;

exports[`Registration test 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    -&gt;
    <span
      className="hidden--thumb"
    >
      Sign in / Sign up
    </span>
  </button>
  <div
    className="relative"
  >
    <div
      style={
        Object {
          "display": "none",
          "height": "100%",
          "left": 0,
          "overflow": "auto",
          "position": "fixed",
          "top": 0,
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
      <div
        onClick={[Function]}
      />
      <div
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 3,
            "boxShadow": "0 20px 30px rgba(0, 0, 0, 0.2)",
            "boxSizing": "border-box",
            "margin": "40px auto",
            "padding": "15px 20px",
            "position": "relative",
            "textAlign": "left",
            "width": 500,
            "zIndex": 2,
          }
        }
      >
        <div
          className="grid--right pb40"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div>
          <button
            onClick={[Function]}
          >
            Login
          </button>
          <button
            onClick={[Function]}
          >
            Register
          </button>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "inherit",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="login_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="login_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="login_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="login_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
        <div
          className="grid pb40"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="1/2 grid__cell"
          >
            <label
              className="grid__cell"
              htmlFor="register_email"
            >
              Email
            </label>
            <label
              className="grid__cell"
              htmlFor="register_username"
            >
              Username
            </label>
            <label
              className="grid__cell"
              htmlFor="register_password"
            >
              Password
            </label>
          </div>
          <div
            className="1/2 grid__cell"
          >
            <input
              className="grid__cell"
              name="register_email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <input
              className="grid__cell"
              name="register_password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="grid--right"
          >
            <button
              onClick={[Function]}
            >
              Send
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
